File: 프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.py
def solution(bandage, health, attacks):


    n=len(attacks)
    a=attacks[n-1][0]

    s=0
    u=0
    real_health=health
    for t in range(a+1):

        if t==0:

            if t==attacks[u][0]:            
                s=0
                real_health=real_health-attacks[u][1]
                u+=1

            continue    

        if t==attacks[u][0]:            
                s=0
                real_health=real_health-attacks[u][1]
                u+=1        


        else:
            if real_health<= health:
                real_health=real_health+bandage[1]
                s+=1

                if real_health>health:
                    real_health=health

                if s==bandage[0]:
                    real_health=real_health+bandage[2]
                    s=0
                    if real_health>health:
                        real_health=health


        if real_health<=0:
            real_health=-1
            break


    answer =real_health
    return answer

File: 프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(vector<int> bandage, int health, vector<vector<int>> attacks) {
    int answer = health;
    int t = 0;
    int bandagecnt = 0;

    for(int time = 1; time<= attacks[attacks.size()-1][0]; time++){
        if(time == attacks[t][0]){
            answer = answer - attacks[t][1];
            t++;
            bandagecnt = 0;

            if(answer<=0) return -1;
        }
        else {
            bandagecnt++;
            answer += bandage[1];

            if(bandagecnt == bandage[0]){
                bandagecnt = 0;
                answer += bandage[2];

            }
            if(answer>health){
                answer = health;
            }
        }
    }

    return answer;
}